cmd/imgcollector/image/hostfs/containerd_image.go

```diff
@@ -42,6 +42,11 @@ func NewContainerdImage(hash v1.Hash, cfg ContainerdHostFSConfig) (Image, error)
 		configBytes: configBytes,
 		contentDir:  cfg.ContentDir,
 	}
+
+	index, err := manifestReader.resolveIndex()
```

Comment:
resolveDigest is called second time here. Maybe it would be better to split it from resolveManifest, and then reuse the same `manifestOrIndex`?